tests/conftest.py

```diff
@@ -6,3 +6,18 @@ def api_client():
     from rest_framework.test import APIClient
 
     return APIClient()
+
+
+@pytest.fixture(autouse=True)
+def force_authenticate(request, api_client):
+    """Automatically authenticate generated requests.
+
+    Check ongoing test for the `as_user` or `as_other_user` marks. To use those marks,
+    `user` and `other_user` must be available in the test scope.
+    """
+    if request.node.get_closest_marker("as_user"):
+        bob = request.getfixturevalue("user")
+        api_client.force_authenticate(bob)
+    elif request.node.get_closest_marker("as_other_user"):
+        alice = request.getfixturevalue("other_user")
```

Comment:
so here, the fixture value is an instance of `User` generated by the test factory ?

Reply:
(typo : ~`bob`~ -> `user`, ~`alice`~ -> `other_user`)

Yes! `request.getfixturevalue("user")` loads the fixture dynamically (when `@pytest.mark.as_user` is decoratin a test)